cmds.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import "github.com/Azure/azure-docker-extension/pkg/vmextension"
+
+type cmdFunc func(vmextension.HandlerEnvironment) error
+
+var (
+	cmds = map[string]struct {
+		f             cmdFunc // associated function
```

Comment:
@boumenot I miss the full argument. the anonymous struct is still type safe. I just use the map to lookup the user-input command e.g. "enable" to the enable method below.